src/transformations/IfElseSwapTransformation.cpp

```diff
@@ -0,0 +1,88 @@
+#include "../../include/transformations/IfElseSwapTransformation.h"
+
+#include <iostream>
+
+using clang::IfStmt, clang::ForStmt, clang::WhileStmt;
+using clang::Lexer, clang::CharSourceRange;
+using llvm::isa, llvm::cast;
+using std::unique_ptr, std::find, std::vector, std::string;
+
+// ------------ IfElseSwapVisitor ------------
+
+IfElseSwapVisitor::IfElseSwapVisitor(Rewriter * rewriter) :
+        rewriter(rewriter), sm(rewriter->getSourceMgr()), opt(rewriter->getLangOpts()) {}
+
+bool IfElseSwapVisitor::VisitStmt(Stmt *s) {
+    if (isa<IfStmt>(s)) {
+        auto ifStmt = cast<IfStmt>(s);
+        if (ifStmt->hasElseStorage()) {
+            auto elseStmt = ifStmt->getElse();
+            // Checking if visitor come to "else if"
+            if ((!isa<IfStmt>(elseStmt))) {
```

Comment:
Let's move this `if-else` into a separate private function. There is a lot of nesting here